### PR TITLE
chore(import-export): remove production dependency on temp COMPASS-7975

### PR DESCRIPTION
**Pull request by mcasimir**
### Checklist
- [ ] New tests and/or benchmarks are included
- [ ] Documentation is changed or added
- [ ] I have signed the MongoDB Contributor License Agreement (https://www.mongodb.com/legal/contributor-agreement)

## Motivation and Context

- [ ] Bugfix
- [ ] New feature
- [ ] Dependency update
- [ ] Misc

## Types of changes

- [ ] *Backport Needed*
- [ ] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Review by addaleax — APPROVED** [review]

File: packages/compass-import-export/src/modules/export.spec.ts [hunk 1]
```diff
@@ -1,15 +1,12 @@
...
 import { expect } from 'chai';
-import temp from 'temp';
 import fs from 'fs';
 import path from 'path';
 import Sinon from 'sinon';
 import type { DataService } from 'mongodb-data-service';
 import { connect } from 'mongodb-data-service';
 import AppRegistry from 'hadron-app-registry';
 
-temp.track();
```
Comment (lerouxb) [inline packages/compass-import-export/src/modules/export.spec.ts:11]: oof. missed this.

File: packages/compass-import-export/src/export/export-csv.ts [hunk 2]
```diff
@@ -223,7 +230,7 @@ async function loadEJSONFileAndColumns({
   // while simultaneously determining the unique set of columns in the order
   // we'll have to write to the file.
   const inputStream = cursor.stream();
-  const filename = temp.path({ suffix: '.jsonl' });
+  const filename = generateTempFilename('.jsonl');
```
Comment (lerouxb) [inline packages/compass-import-export/src/export/export-csv.ts:233]: This package depends on `temp` already, why not just use that?

File: packages/compass-import-export/package.json [hunk 3]
```diff
@@ -106,6 +105,7 @@
     "react-dom": "^17.0.2",
     "sinon": "^9.2.3",
     "sinon-chai": "^3.7.0",
+    "temp": "^0.9.4",
```
Comment (lerouxb) [inline packages/compass-import-export/package.json:108]: Makes sense. Should always have been a dev dep.

**Review by lerouxb — APPROVED** [review]

File: packages/compass-import-export/src/export/export-csv.ts (hunk 2 above)
Comment (mcasimir) [reply packages/compass-import-export/src/export/export-csv.ts:233]: it brings 1000 dependencies and has a vulnerability

**Pull request merged**